src/Loggers/LogImage.jl

```diff
@@ -1,38 +1,191 @@
+    @assert isa(first(imgArrays), AbstractArray{<:Colorant}) "Please specify format"
+    for (n, imgArray) in enumerate(imgArrays)
+        log_image(logger, name*"/$n", imgArray, step = step)
+    end
+end
+
 function log_images(logger::TBLogger, name::AbstractString, imgArrays::AbstractArray, format::ImageFormat; step = nothing)
-    n = 1
-    for imgArray in imgArrays
+    for (n, imgArray) in enumerate(imgArrays)
         log_image(logger, name*"/$n", imgArray, format, step = step)
-        n += 1
     end
 end
 """
     log_image(logger::TBLogger, name::AbstractString, imgArray::AbstractArray, format::ImageFormat, step = nothing)
 
 Log an image using image data and format
-- imgArray: image data. A 2-D or 3-D `Array` of pixel values. pixel values can be Real [0, 1] or Integer[0, 255]
-- format: format of the image. It can be one of {HW, WH, HWC, WHC, CHW, CWH, HWN, WHN, NHW, NWH, HWCN, WHCN, CHWN, CWHN, NHWC, NWHC, NCHW, NCWH}
+- imgArray: image data. A 1-D, 2-D or 3-D `Array` of pixel values. pixel values can be Real [0, 1] or Integer[0, 255]
+- format: format of the image. It can be one of
+  {L, LN, NL,
+   CL, LC, NCL, NLC, CLN, LCN,
+   HW, WH, HWC, WHC, CHW, CWH,
+   HWN, WHN, NHW, NWH,
+   HWCN, WHCN, CHWN, CWHN, NHWC, NWHC, NCHW, NCWH}
+  - L: Length
+  - C: Channel/Color
+  - H: Height
+  - W: Width
+  - N: Observation
 """
+function log_image(logger::TBLogger, name::AbstractString, img::AbstractArray{<:Colorant}; step = nothing)
+    dimensions = ndims(img)
+    #if it has 3 dimensions, eg MRI image, log each channel vth same name
+    #so that one can slide through channels
+    if dimensions == 3
+        #3rd d is channel dim as observed in testimages
+        channels = size(img, 3)
+        for c in 1:channels-1
+            log_image(logger, name, img[:, :, c], step = step)
+        end
+        img = img[:, :, channels]
+    end
+    summ = SummaryCollection()
+    push!(summ.value, image_summary(name, img))
+    write_event(logger.file, make_event(logger, summ, step=step))
+end
+
 function log_image(logger::TBLogger, name::AbstractString, imgArray::AbstractArray, format::ImageFormat; step=nothing)
+    summ = SummaryCollection()
+    #passing logger and step to image_summary because,
+    #format containing N will have to use them to call `log_image`
+    push!(summ.value, image_summary(name, imgArray, format, logger = logger, step = step))
+    write_event(logger.file, make_event(logger, summ, step=step))
+end
+
+function image_summary(name::AbstractString, img::AbstractArray{<:Colorant})
+    #image is of type AbstractArray{<:Colorant}
+    dimensions = ndims(img)
+    if isa(first(img), Gray)
+        colorspace = 1
+    elseif isa(first(img), GrayA)
+        colorspace = 2
+    elseif isa(first(img), RGB)
+        colorspace = 3
+    elseif isa(first(img), RGBA)
+        colorspace = 4
+    elseif isa(first(img), BGRA)
+        colorspace = 6
+    else
+        throw("Unknown Colorspace")
+    end
```

Comment:
This could be done by defining a method that dispatches:

like

```
colorspace_id(::Type{Gray}) = 1
colorspace_id(::Type{GrayA}) = 2
...
function image_summary(name::AbstractString, img::AbstractArray{<:Colorant})
     ...
    colorspace = colorspace_id(eltype(img))
    ...
end
```